main.cpp

```diff
@@ -76,6 +76,101 @@ returning lhs
 10
 good to go!
 */
+#include <iostream>
+struct FloatType
+{
+    //float lhs, rhs;
+    
+    float add( float lhs, float rhs )
+    {
+        return lhs + rhs;
+    }
+    float subtract( float lhs, float rhs )
+    {
+        return lhs - rhs;
+    }
+
+    float multiply( float lhs, float rhs )
+    {
+        return lhs * rhs;
+    }
+    float divide( float lhs, float rhs )
+    {
+        if(rhs == 0.0f)
+        { 
+            std::cout << "\nwarning error, integer division by zero returns 'inf' !"  << std::endl;
+        }
+        
+        return lhs / rhs;
+    }
+
+
+
+};
+
+struct DoubleType
+{
+        
+    double add( double lhs, double rhs )
+    {
+        return lhs + rhs;
+    }
+    double subtract( double lhs, double rhs )
+    {
+        return lhs - rhs;
+    }
+    double multiply( double lhs, double rhs )
+    {
+        return lhs * rhs;
+    }
+    double divide( double lhs, double rhs )
+    {
+         if(rhs == 0.0)
+        { 
+            std::cout << "\nwarning error, integer division by zero returns 'inf' !"  << std::endl;
+        }
+        
+        return lhs / rhs;
+    }
+
+};
+
+struct IntType
+{
+   
+    int add( int lhs, int rhs )
+    {
+        return lhs + rhs;
+    }
+    int subtract( int lhs, int rhs )
+    {
+        return lhs - rhs;
+    }
+    int multiply( int lhs, int rhs )
+    {
+        return lhs * rhs;
+    }
+    int divide( int lhs, int rhs )
+    {
+        if(rhs == 0.0f)
+        { 
+            std::cout << "\nwarning error, integer division by zero will crash the program!"  << std::endl;
+            std::cout << "returning lhs" << std::endl << lhs << std::endl;
+            return 0;
```

Comment:
You aren't returning `lhs`, like the output message says.
- [ ] revise

Reply:
ah, that was me lacking insight into the task than carelessness, (I carefully printed lhs!). I get what you were asking for now.